docs/migration_guide.md

```diff
@@ -0,0 +1,127 @@
+```
+apiVersion: sdv.eclipse.org/v1
+kind: SelfUpdateBundle
+metadata: 
+    name: "self-update-bundle-example"
+spec: 
+    bundleVersion: 1.0
+```
+
+```
+{
+    "timestamp": 42,
+    "payload": {
+        "softwareNodes": [
+            {
+                "id": "self-update-agent",
+                "version": "build-42",
+                "name": "OTA NG Self Update Agent",
+                "type": "APPLICATION"
+            },
+            {
+                "id": "self-update:leda-deviceimage",
+                "version": "1.0",
+                "name": "Official Leda device image",
+                "type": "IMAGE"
+            }
+        ],
+        "hardwareNodes": [],
+        "associations": [
+            {
+                "sourceId": "self-update-agent",
+                "targetId": "self-update:leda-deviceimage"
+            }
+        ]
+    }
+}
+```
+
+### Transition guide for desired state message
+For the desired state message there are 2 relevant fields 'bundleDownloadUrl' and 'bundleVersion' for take-over. The corresponding values are 'version' under the 'components' and 'value' inside the object with 'key'='image'. *NOTE*: There could be multiple entries inside 'components' array and 'config' section. Your implementation has to search for 'os-image' and 'image' values respectively.
+```
+apiVersion: "sdv.eclipse.org/v1"
+kind: SelfUpdateBundle
+metadata:
+    name: self-update-bundle-example
+spec:
+    bundleDownloadUrl: http://url
+    bundleName: arm64-bundle
+    bundleTarget: base
+    bundleVersion: 1.0
+```
+```
+{
+    "activityId": "random-uuid-as-string",
+    "timestamp": 123456789,
+    "payload": {
+        "domains": [
+            {
+                "id": "self-update",
+                "components": [
+                    {
+                        "id": "os-image",
+                        "version": "1.1",
+                        "config": [
+                            {
+                                "key": "image",
+                                "value": "http://example.com/downloads/os-image-1.1.bin"
+                            }
+                        ]
+                    }
+                ]
+            }
+        ]
+    }
+}
+```
+
+### Transition guide for state feedback
+Below the example for state feedback ('techCode' is optional and available only in case of failure). For transition to JSON format all 3 fields from 'state' section are important (except 'techCode') and their corresponding places in JSON are 'status' and 'message' in secions 'payload' and 'actions', and 'progress' in section 'actions'. Detailed list of available payload status and action status can be found in [link](docs/bfb.md) because for vehicles orchestration more sub-states were introduces for fine-grained reporting of self-update progress.
```

Comment:
"Below there is an example...",
"... all three fields ...",
"The detailed list of the available...",
typo: "... in sections 'payload'", 
"... were introduced ...", 
"... reporting of the self-update progress."